controllers/datadogagent/agent_rbac.go

```diff
@@ -65,6 +65,10 @@ func (r *Reconciler) manageAgentRBACs(logger logr.Logger, dda *datadoghqv1alpha1
 		return reconcile.Result{}, err
 	}
 
+	if result, err := r.udpateIfNeededAgentClusterRoleBinding(logger, dda, rbacResourcesName, serviceAccountName, agentVersion, clusterRoleBinding); err != nil {
+		return result, err
+	}
+
 	return reconcile.Result{}, nil
```

Comment:
like this the result will be never used if `err == nil` 
I think since it is the last action in this function you can do directly:

```suggestion
	return r.udpateIfNeededAgentClusterRoleBinding(logger, dda, rbacResourcesName, serviceAccountName, agentVersion, clusterRoleBinding)
```